dashboard/app/models/user.rb

```diff
@@ -1308,4 +1308,25 @@ def show_school_info_interstitial?
   def school_info_suggestion?
     !(school.blank? && full_address.blank?)
   end
+
+  # Removes PII and other information from the user and marks the user as having been purged.
+  # WARNING: This (permanently) destroys data and cannot be undone.
+  # WARNING: This does not purge the user, only marks them as such.
+  def clear_user_and_mark_purged
+    self.name = nil
+    self.username = 'abcdef'
```

Comment:
As discussed offline, because there is a unique index on `(users.username, users.deleted_at)`, we don't want to use a fixed username. Though the implementation is not the cleanest, it serves both to indicate the user was purged (via the username), prevent DB uniqueness violations, and prevent Rails validation violations (otherwise, a longer random string could be used).